DQM/SiTrackerPhase2/plugins/Phase2ITMonitorCluster.cc

```diff
@@ -0,0 +1,416 @@
+  edm::EDGetTokenT<edmNew::DetSetVector<SiPixelCluster>> itPixelClusterToken_;
+  const edm::ESGetToken<TrackerGeometry, TrackerDigiGeometryRecord> geomToken_;
+  const edm::ESGetToken<TrackerTopology, TrackerTopologyRcd> topoToken_;
+  const TrackerGeometry* tkGeom_ = nullptr;
+  const TrackerTopology* tTopo_ = nullptr;
+};
+#endif
+
+#include "DQM/SiTrackerPhase2/interface/TrackerPhase2DQMUtil.h"
+
+//
+// constructors
+//
+Phase2ITMonitorCluster::Phase2ITMonitorCluster(const edm::ParameterSet& iConfig)
+    : config_(iConfig),
+      itPixelClusterToken_(consumes<edmNew::DetSetVector<SiPixelCluster>>(
+          config_.getParameter<edm::InputTag>("InnerPixelClusterSource"))),
+      geomToken_(esConsumes<TrackerGeometry, TrackerDigiGeometryRecord, edm::Transition::BeginRun>()),
+      topoToken_(esConsumes<TrackerTopology, TrackerTopologyRcd, edm::Transition::BeginRun>()) {
+  edm::LogInfo("Phase2ITMonitorCluster") << ">>> Construct Phase2ITMonitorCluster ";
+}
+
+Phase2ITMonitorCluster::~Phase2ITMonitorCluster() {
+  edm::LogInfo("Phase2ITMonitorCluster") << ">>> Destroy Phase2ITMonitorCluster ";
+}
+//
+// -- DQM Begin Run
+void Phase2ITMonitorCluster::dqmBeginRun(const edm::Run& iRun, const edm::EventSetup& iSetup) {
+  edm::ESHandle<TrackerGeometry> geomHandle = iSetup.getHandle(geomToken_);
+  tkGeom_ = &(*geomHandle);
+  edm::ESHandle<TrackerTopology> tTopoHandle = iSetup.getHandle(topoToken_);
+  tTopo_ = tTopoHandle.product();
+}
+
+//
+// -- Analyze
+//
+void Phase2ITMonitorCluster::analyze(const edm::Event& iEvent, const edm::EventSetup& iSetup) {
+  // Getting the clusters
+  edm::Handle<edmNew::DetSetVector<SiPixelCluster>> itPixelClusterHandle;
+  iEvent.getByToken(itPixelClusterToken_, itPixelClusterHandle);
+  // Number of clusters
+  std::map<std::string, unsigned int> nClsmap;
+  unsigned int nclusGlobal = 0;
+  for (const auto& DSVItr : *itPixelClusterHandle) {
+    uint32_t rawid(DSVItr.detId());
+    DetId detId(rawid);
+    const GeomDet* geomDet = tkGeom_->idToDet(detId);
+    if (!geomDet)
+      continue;
+    const GeomDetUnit* geomDetUnit(tkGeom_->idToDetUnit(detId));
+    if (!geomDetUnit)
+      continue;
+    nclusGlobal += DSVItr.size();
+    std::string folderkey = phase2tkutil::getITHistoId(detId, tTopo_);
+    // initialize the nhit counters if they don't exist for this layer
+    if (nClsmap.find(folderkey) == nClsmap.end()) {
+      nClsmap.emplace(folderkey, 0);
+    }
+    nClsmap.at(folderkey) += DSVItr.size();
+    for (const auto& clusterItr : DSVItr) {
+      MeasurementPoint mpCluster(clusterItr.x(), clusterItr.y());
+      Local3DPoint localPosCluster = geomDetUnit->topology().localPosition(mpCluster);
+      Global3DPoint globalPosCluster = geomDetUnit->surface().toGlobal(localPosCluster);
+      double gx = globalPosCluster.x() * 10.;
+      double gy = globalPosCluster.y() * 10.;
+      double gz = globalPosCluster.z() * 10.;
+      double gr = globalPosCluster.perp() * 10.;
```

Comment:
why the unit conversion?

Reply:
It has been the preferred choice from our side to have the position maps in mm. All the other modules also have the conversions.